### PR TITLE
[WIP]support scheduler configurable

### LINKED ISSUE #281: make scheduler plugins configurable
#### What would you like to be added:
- configure `clusternet-scheduler` with a profile, where the scheduling plugins could be configured
- each profile has an associated scheduler name and can have a different set of plugins configured in its extension points

Referring [kube-scheduler profiles](https://kubernetes.io/docs/reference/scheduling/config/#profiles)

#### Why is this needed:
A scheduling Profile allows you to configure the different stages of scheduling in the `clusternet-scheduler`. Each stage is exposed in an extension point. Plugins provide scheduling behaviors by implementing one or more of these extension points.

### PR DESCRIPTION
**Pull request by silenceper**
#### What type of PR is this?
- support scheduler configurable
- support add custom plugin 

#### Which issue(s) this PR fixes:

Fixes #281

### CODE REVIEW

File: pkg/scheduler/apis/config/scheme/scheme.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	Codecs = serializer.NewCodecFactory(Scheme, serializer.EnableStrict)
+)
+
+func init() {
+	AddToScheme(Scheme)
+}
+
+// AddToScheme builds the scheduler scheme using all known versions of the scheduler api.
+func AddToScheme(scheme *runtime.Scheme) {
+	utilruntime.Must(schedulerconfig.AddToScheme(scheme))
```
Comment (dixudx) [inline pkg/scheduler/apis/config/scheme/scheme.go:41]: Currently we don't need have such an internal scheme.

File: pkg/scheduler/apis/config/scheme/scheme.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	schedulerconfig "github.com/clusternet/clusternet/pkg/scheduler/apis/config"
+	schedulerconfigv1alpha1 "github.com/clusternet/clusternet/pkg/scheduler/apis/config/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+var (
+	// Scheme is the runtime.Scheme to which all scheduler api types are registered.
+	Scheme = runtime.NewScheme()
```
Comment (dixudx) [inline pkg/scheduler/apis/config/scheme/scheme.go:29]: We may not need to register this scheme. It is neither for a REST api, nor for a CRD.

**Review by dixudx — CHANGES_REQUESTED** [review]
Please reformat all the go imports in this PR as well.

File: pkg/scheduler/apis/config/scheme/scheme.go (hunk 2 above)
Comment (silenceper) [reply pkg/scheduler/apis/config/scheme/scheme.go:29]: This is to parse a configuration file like the following:
```
apiVersion: scheduler.config.clusternet.io/v1alpha1
kind: SchedulerConfiguration
profiles:
  -  plugins:
      predict:
        enabled:
        - name: "testPlugin"
        disabled:
        - name: "*"
      assign:
        enabled:
        - name: 'testPlugin'
        disabled:
        - name: '*'
      bind:
        enabled:
        - name: 'testPlugin'
        disabled:
        - name: '*'
```

like kube-scheduler: https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/scheme/scheme.go

**Comment by silenceper** [comment]
@dixudx  I used the goimport tool to format, but nothing changed, are there any rules?

File: pkg/scheduler/apis/config/scheme/scheme.go (hunk 2 above)
Comment (dixudx) [reply pkg/scheduler/apis/config/scheme/scheme.go:29]: It works with `json.Unmarshal`, since we know the exact scheme for deserialziation.

**Comment by dixudx** [comment]
> @dixudx I used the goimport tool to format, but nothing changed, are there any rules?

built-in packages
third-party packages
local repo packages

**Comment by silenceper** [comment]
> > @dixudx I used the goimport tool to format, but nothing changed, are there any rules?
> 
> built-in packages third-party packages local repo packages

consider add this rule to ci check?

**Comment by dixudx** [comment]
> > > @dixudx I used the goimport tool to format, but nothing changed, are there any rules?
> > 
> > 
> > built-in packages third-party packages local repo packages
> 
> consider add this rule to ci check?

Of course. Thanks for pointing it out. I've added a linter in #397.

**Pull request closed without merging**